capstone/src/main/java/com/google/sps/servlets/YoutubeKeywordServlet.java

```diff
@@ -0,0 +1,61 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+import java.io.BufferedReader;
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import org.json.simple.JSONObject;    
+import java.util.ArrayList;
+import java.util.Arrays;
+
+/** Servlet that returns youtube video data based on a keyword. */
+@WebServlet("/keyword_search_servlet")
+public class YoutubeKeywordServlet extends HttpServlet {
+  private static final String BASE_URL = "https://www.googleapis.com/youtube/v3/search?part=snippet";
+  private static final String KEY = "API_KEY";
+  private static final String NUM_RESULTS = "5";
+  private static final MediaType JSON = MediaType.get("application/json; charset=utf-8");
+  OkHttpClient client = new OkHttpClient();
+  
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String searchTerm = request.getParameter("searchTerm");
+    String completeUrl = BASE_URL + "&maxResults=" + NUM_RESULTS + "&q=" + searchTerm + "&key=" + KEY;
+    String output = get(completeUrl);
+    response.setContentType("application/json");
+    response.getWriter().println(output);  
+  }
+  
+  /** Makes a GET request. */
+  private String get(String url) throws IOException {
```

Comment:
I think this function exists now in a few different places - is it possible to refactor this code so it's only defined in 1 .java file?

Reply:
Yes, that is certainly possible. I plan to make these changes in a separate PR.